purchases/src/main/kotlin/com/revenuecat/purchases/common/events/BackendStoredEvent.kt

```diff
@@ -0,0 +1,90 @@
+package com.revenuecat.purchases.common.events
+
+import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI
+import com.revenuecat.purchases.paywalls.events.PaywallEvent
+import com.revenuecat.purchases.utils.Event
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+/**
+ * Stored backend events to be flushed together by EventsManager.
+ */
+@Serializable
+internal sealed class BackendStoredEvent : Event {
+
+    /**
+     * Represents a stored event related to the Customer Center.
+     *
+     * @property event The `BackendEvent.CustomerCenter` event that is being stored.
+     */
+    @Serializable
+    @SerialName("customer_center")
+    data class CustomerCenter(val event: BackendEvent.CustomerCenter) : BackendStoredEvent()
+
+    /**
+     * Represents a stored event related to Paywalls.
+     *
+     * @property event The `BackendEvent.Paywalls` event that is being stored.
+     */
+    @Serializable
+    @SerialName("paywalls")
+    data class Paywalls(val event: BackendEvent.Paywalls) : BackendStoredEvent()
+}
+
+/**
+ * Converts a `BackendStoredEvent` into a `BackendEvent`.
+ *
+ * @receiver The stored backend event to be converted.
+ * @return A `BackendEvent` instance or `null` if conversion is not possible.
+ */
+@OptIn(ExperimentalPreviewRevenueCatPurchasesAPI::class)
+internal fun BackendStoredEvent.toBackendEvent(): BackendEvent? {
+    return when (this) {
+        is BackendStoredEvent.Paywalls -> {
```

Comment:
this can be now `is BackendStoredEvent.Paywalls -> { this.event }` since it's already a `BackendEvent.Paywalls`. You might not even need `toBackendEvent` at all? I would at least rename it to indicate it returns the underlying event and it's not a conversion